src/main/java/org/junit/rules/TemporaryFolder.java

```diff
@@ -167,16 +169,22 @@ public File newFile() throws IOException {
     }
 
     /**
-     * Returns a new fresh folder with the given name under the temporary
+     * Returns a new fresh folder with the given path under the temporary
      * folder.
      */
     public File newFolder(String folder) throws IOException {
-        return newFolder(new String[]{folder});
+        if (folder.startsWith(File.separator)) {
+            throw new IOException("Folder name cannot start with a file separator");
+        }
+        return newFolder(folder.split(FILE_SEPARATOR_REGEX));
```

Comment:
I didn't realize that File supported / as a path separator for Windows.

Instead of your solution, how about I rewite this method to use mkdirs()?